libs/community/langchain_community/chat_models/baidu_qianfan_endpoint.py

```diff
@@ -512,22 +512,30 @@ def _generate(
         if self.streaming:
             completion = ""
             chat_generation_info: Dict = {}
+            usage_metadata: Optional[UsageMetadata] = None
             for chunk in self._stream(messages, stop, run_manager, **kwargs):
                 chat_generation_info = (
                     chunk.generation_info
                     if chunk.generation_info is not None
                     else chat_generation_info
                 )
                 completion += chunk.text
-            lc_msg = AIMessage(content=completion, additional_kwargs={})
+                if isinstance(chunk.message, AIMessageChunk):
+                    usage_metadata = chunk.message.usage_metadata
+
+            lc_msg = AIMessage(
+                content=completion,
+                additional_kwargs={},
+                usage_metadata=usage_metadata,
```

Comment:
it looks like `usage_metadata` will be the last value from the loop. is it typically present in the last value? (e.g., so it won't get overwritten with None during the loop)